### ISSUE TITLE
Cannot install horovod with tensorflow framework

### ISSUE THREAD

**Issue by Yikai-coder** [horovod/horovod#3383]
I was trying to install horovod using the command below:
``` HOROVOD_WITHOUT_GLOO=1 HOROVOD_WITHOUT_MXNET=1 HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_OPERATIONS=NCCL HOROVOD_NCCL_HOME=/home/li/nccl/build pip install --no-cache-dir horovod[tensorflow] ```
  After a while, I saw that the horovod was successfully installed, but when I run ```horovodrun --check-build``` to check the installation, I did not see the tensorflow framework was installed.
```
Horovod v0.23.0:

Available Frameworks:
    [ ] TensorFlow
    [X] PyTorch
    [ ] MXNet

Available Controllers:
    [X] MPI
    [ ] Gloo

Available Tensor Operations:
    [X] NCCL
    [ ] DDL
    [ ] CCL
    [X] MPI
    [ ] Gloo 
```
When I tried to import the module by ```import horovod.tensorflow```, I got such error:
```
tensorflow.python.framework.errors_impl.NotFoundError: /home/li/.local/lib/python3.8/site-packages/horovod/tensorflow/mpi_lib.cpython-38-x86_64-linux-gnu.so: undefined symbol: cudaGetDriverEntryPoint, version libcudart.so.11.0
```
The environment I use now is below:
```
Ubuntu 20.04
CUDA: 11.6
python: 3.8.10
Horovod: 0.23.0
tensorflow 2.7.0
```
I will be sincerely gratefully if someone can help me fix this problem.

**Comment by maxhgerlach** [comment]
Can you install and then import tensorflow independently of Horovod? Does TF really support CUDA 11.6?

**Comment by Yikai-coder** [comment]
Yes, I can install and import tensorflow independently and also run my program on a single GPU with CUDA 11.6.

**Comment by maxhgerlach** [comment]
I see, thanks for reporting back. In my experience TF on GPU really depends on having just the right version of the CUDA Toolkit available and, according to https://www.tensorflow.org/install/gpu, that's CUDA Toolkit 11.2 for TF >= 2.5 at this time. PyTorch seems to be less picky, I think because they ship statically linked CUDA libraries.

Generally it seems to be safest to first install your desired version of TensorFlow, then install Horovod in a separate `pip install` command, but I don't think that's the problem here.

**Comment by maxhgerlach** [comment]
Here you can see which version of CUDA Toolkit and cuDNN you need for each version of TensorFlow: https://www.tensorflow.org/install/source#gpu

**Comment by Yikai-coder** [comment]
Thank you for helping, I have just successfully installed horovod with the correct framework, thought I didn't know why. Here are the process.
I first tried to install without NCLL support, by `HOROVOD_WITH_TENSORFLOW pip install horovod[tensorflow] --no-cache-dir`, and I found that horovod was installed with Tensorflow framework support correctly, but without NCLL support. 
So I tried to add one more argument, using `HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_OPERATIONS=NCCL HOROVOD_NCCL_HOME=/home/li/nccl/build pip install horovod[tensorflow] --no-cache-dir`. But then I got an error compiling,     reminding me that the MXNET in my computer didn't support CUDA. I always add `HOROVOD_WITHOUT_MXNET` to skip this error, but this time I tried to figure it out, by installing MXNET again.
After successfully installing MXNET, I tried to install horovod again, and didn't get any error this time. But when I check the build, I found that all the framework support was not installed. So I uninstall horovod and tried to install it again. After a few times of try, maybe 3 or 4 times, I found that I got tht correct horovod installed.
Anyway, I have now successfully installed horovod, and above are my experience of installing. Hope that can help to fix some problems and thank you for helping.
